redwood-lazylayout-compose/src/commonMain/kotlin/app/cash/redwood/lazylayout/compose/LazyList.kt

```diff
@@ -46,9 +46,8 @@ internal fun LazyList(
 ) {
   val itemProvider = rememberLazyListItemProvider(content)
   var lastVisibleItemIndex by remember { mutableStateOf(0) }
-  val itemsBefore = remember(state.firstVisibleItemIndex) { (state.firstVisibleItemIndex - OffscreenItemsBufferCount / 2).coerceAtLeast(0) }
+  val itemsBefore = remember(state.scrollItemIndex) { (state.scrollItemIndex - OffscreenItemsBufferCount / 2).coerceAtLeast(0) }
```

Comment:
ooooh I think we might have broken this. 

This should probably be a different mutableState for proper on-demand loading

Reply:
As-is this only updates once, and I think it needs to update with every scroll.

Reply:
done